tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs

```diff
@@ -469,8 +467,45 @@ void TestJsonDeserializationCreatesJavaHandle ()
 			if (p2.Handle == IntPtr.Zero)
 				throw new InvalidOperationException (""Failed to instantiate new Java instance for Person!"");
 
-			Console.WriteLine ($""JSON Person deserialized OK"");").Replace ("//${AFTER_MAINACTIVITY}", @"
+			Console.WriteLine ($""JSON Person deserialized OK"");
+		}
+
+		void TestBinaryDeserialization (Person p)
+		{
+			var stream      = new MemoryStream ();
+			var serializer  = new BinaryFormatter();
+
+			serializer.Serialize (stream, p);
+
+			stream.Position = 0;
+			StreamReader sr = new StreamReader (stream);
+
+			stream.Position = 0;
+			Person p2 = (Person) serializer.Deserialize (stream);
+
+			Console.WriteLine ($""BinaryFormatter deserialzied: Name '{p2.Name}' Age '{p2.Age}' Handle '0x{p2.Handle:X}'"");
+
+			if (p2.Name != ""John Smith"")
+				throw new InvalidOperationException (""BinaryFormatter deserialization of Name"");
+			if (p2.Age != 900)
+				throw new InvalidOperationException (""BinaryFormatter deserialization of Age"");
+			if (p2.Handle == IntPtr.Zero)
+				throw new InvalidOperationException (""Failed to instantiate new Java instance for Person!"");
+
+			Console.WriteLine ($""BinaryFormatter Person deserialized OK"");
+		}
+
+		void TestJsonDeserializationCreatesJavaHandle ()
```

Comment:
This added test is failing under One .NET: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4365170&view=ms.vss-test-web.build-test-results-tab&runId=18051660&resultId=100089&paneView=attachments

Reply:
Nice. The net6 is more strict about unsafe binary serialization. I have added the binder and also disabled it in release as it fails with illink as linking of serialization is still broken on net6.